flocker/node/agents/ebs.py

```diff
@@ -195,11 +300,71 @@ def resize_volume(self, blockdevice_id, size):
         pass
 
     # cloud_instance_id here too
-    def attach_volume(self, blockdevice_id, host):
-        pass
+    def attach_volume(self, blockdevice_id, attach_to):
+        """
+        Attach an EBS volume to given compute instance.
+
+        :param unicode blockdevice_id: EBS UUID for volume to be attached.
+        :param unicode attach_to: Instance id of AWS Compute instance to
+            attached the blockdevice to.
+
+        :raises UnknownVolume: If there does not exist a BlockDeviceVolume
+            corresponding to the input blockdevice_id.
+        :raises AlreadyAttachedVolume: If the input volume is already attached
+            to a device.
+        """
+        volume = self._get(blockdevice_id)
+        if volume.attached_to is not None:
+            raise AlreadyAttachedVolume(blockdevice_id)
+
+        self.lock.acquire()
```

Comment:
I think you can use `with` with this lock object.  It's a "context manager".  This is syntactic sugar for doing the `release` in the equivalent of a `finally` block at the end of a try/finally.  So this makes the code a bit clearer (you get an indented block for all of the code run while the lock is held) and safer (if an exception is raised the lock is released for you) and shorter (you can delete the explicit release because `with` will do it for you).